hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java

```diff
@@ -111,12 +127,51 @@ public interface PreHandleContext {
     @NonNull
     PreHandleContext requireKey(@NonNull final Key key);
 
+    /**
+     * Adds the given key to optional non-payer keys.
+     * If the key is the same as the payer key, or if the key has already been added, then the call
+     * is a no-op. The key must not be null.
+     *
+     * @param key key to be added
+     * @return {@code this} object
+     * @throws NullPointerException if the key is null
+     */
+    @NonNull
+    PreHandleContext optionalKey(@NonNull final Key key);
+
+    /**
+     * Adds the given set of keys to optional non-payer keys.
+     * If any key is the same as the payer key, or if any key has already been added, then the call
+     * ignores that key. The set of keys must not be null, but may be empty.
+     *
+     * @param keys the set of keys to be added
+     * @return {@code this} object
+     * @throws NullPointerException if the set of keys is null
+     */
+    @NonNull
+    PreHandleContext optionalKeys(@NonNull final Set<Key> keys);
+
+    /**
+     * Adds the given hollow account to the optional signing set.
+     * If the account has already been added, then the call is a no-op. The account must not be null.
```

Comment:
```suggestion
     * If the account is invalid or has already been added, then the call is a no-op. The account must not be null.
```

Reply:
We don't actually check for valid in this call, as it requires state access that isn't available to PreHandle context.  We do check if the account is hollow, but that throws an exception (and is documented below).